### PR TITLE
PokePointer and TouchableVolume fix for packed scenes and nested set-ups

### PR DESCRIPTION
**Pull request by jloehr**
## Overview

As described in the issue, when touching a volume, a long distance raycast through the entire scene was used for the poke pointer. This long distance raycast was often hitting other colliders in more packed scenes or certain set-ups.

This change introduces proper penetration distance calculation for touched `NearInteractionTouchableVolume`. This in turn allows for more narrow raycasting, raising the chance of hitting the correct object in the scene.

With the correct distance to the surface of a touched volume, we can also now properly sort touched volumes, resolving intersecting and nested touchable volumes.

Video of the fix in action:

https://user-images.githubusercontent.com/4631548/162210527-acb0a12f-5ec9-4d49-b7ff-d1b3047f1e0e.mp4

## Changes
- Fixes: #10530

### CODE REVIEW

File: Assets/MRTK/Services/InputSystem/NearInteractionTouchableVolume.cs [hunk 1]
```diff
@@ -52,9 +52,21 @@ public override float DistanceToTouchable(Vector3 samplePoint, out Vector3 norma
...
                 normal.Normalize();
-                // Return value less than zero so that when poke pointer is inside
-                // object, it will not raise a touch up event.
-                return -1;
+
+                // Try to calculate the proper penetration distance, to allow more accurate processing of touchable volumes.
+                // Return value less than zero so that when poke pointer is inside object, it will not raise a touch up event.
+                float rayScale = 1.1f;
+                Vector3 outsidePoint = TouchableCollider.bounds.center + normal * (TouchableCollider.bounds.extents.magnitude * rayScale);
+                if (TouchableCollider.Raycast(new Ray(outsidePoint, -normal), out RaycastHit raycastHit, TouchableCollider.bounds.size.magnitude * rayScale))
```
Comment (RogPodge) [inline Assets/MRTK/Services/InputSystem/NearInteractionTouchableVolume.cs:60]: I have some concerns about performing a raycast everything this calculation is done. Is there a way to calculate the distance using the existing normal vector?

**Review by RogPodge — CHANGES_REQUESTED** [review]
Initial check seems good. I need some time to try it out myself and investigate if there's a way to do this without the additional performance hit of another raycast. 

@jloehr do you have time to revisit this? Otherwise, do I have permission to clone and implement any changes that might be required? I really appreciate you contributing this valuable fix (and including unit tests!!)

File: Assets/MRTK/Services/InputSystem/NearInteractionTouchableVolume.cs (hunk 1 above)
Comment (jloehr) [reply Assets/MRTK/Services/InputSystem/NearInteractionTouchableVolume.cs:60]: I tried to search for other ways, but didn't find any. 
TouchableCollider ususally is a `BoxCollider`, but can also be any arbitrary collider including a complex mesh collider. So doing any math based upon that assumption basically resolves to a raycast.
Also keep in mind, this raycast is only executed if we're already touching a volume and only for volumes we're touching. Further it's **NOT** a `Physics.Raycast` but a `Collider.Raycast` which is only considering the given collider. Therefore should be hopefully cheap (especially if its a primitive collider).

**Comment by RogPodge** [comment]
> @RogPodge I do have some spare time to revisit this, so hit me up with suggestions.

Thanks! Can you address the first PR comment I left? I'll add more as I get chances to dig through the PR.

**Comment by RogPodge** [comment]
Also a heads up, can you retarget this to https://github.com/microsoft/MixedRealityToolkit-Unity/tree/prerelease/2.8.0, since we're going to be cutting a release soon and would really like to get this in!

**Comment by jloehr** [comment]
> Also a heads up, can you retarget this to https://github.com/microsoft/MixedRealityToolkit-Unity/tree/prerelease/2.8.0, since we're going to be cutting a release soon and would really like to get this in!

You mean, retargeting this PR to merge into the 2.8.0 prelease branch, right?

**Comment by jloehr** [comment]
> Also a heads up, can you retarget this to https://github.com/microsoft/MixedRealityToolkit-Unity/tree/prerelease/2.8.0, since we're going to be cutting a release soon and would really like to get this in!

Done! Surprisingly no merge conflicts and tests are still fine on my local system.

**Review by RogPodge — APPROVED** [review]

**Pull request merged**